backend/service/k8s/event.go

```diff
@@ -0,0 +1,61 @@
+package k8s
+
+import (
+	"context"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sapiv1 "github.com/lyft/clutch/backend/api/k8s/v1"
+)
+
+func (s *svc) ListEvents(ctx context.Context, clientset, cluster, namespace, object string, kind k8sapiv1.ObjectKind) ([]*k8sapiv1.Event, error) {
+	cs, err := s.manager.GetK8sClientset(ctx, clientset, cluster, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	objKind := strcase.ToCamel(strings.ToLower(kind.String()))
+	// returns the appropriate field selector based on the object involved
+	fieldSelector := cs.CoreV1().Events(cs.Namespace()).GetFieldSelector(&object, &namespace, &objKind, nil)
+
+	eventList, err := cs.CoreV1().Events(cs.Namespace()).List(ctx, metav1.ListOptions{FieldSelector: fieldSelector.String()})
+	if err != nil {
+		return nil, err
+	}
+
+	var events []*k8sapiv1.Event
+	for _, e := range eventList.Items {
+		event := e
+		events = append(events, ProtoForEvent(cs.Cluster(), &event))
+	}
```

Comment:
```suggestion
	for _, e := range eventList.Items {
		events = append(events, ProtoForEvent(cs.Cluster(), &e))
	}
```

Reply:
Changed this to be
```
for i, _ := range eventList.Items {
		events = append(events, ProtoForEvent(cs.Cluster(), &eventList.Items[i]))
	}
```
Since Go uses a copy of the value instead of the value itself within a range clause when we take the pointer of value, we’re actually taking the pointer of a copy of the value. This copy gets reused throughout the range clause, so its safer to use the index to take a pointer to the actual value, instead of the copy.

Reply:
Increase the test coverage to test for this case.

Reply:
Added more than 1 event the test: https://github.com/lyft/clutch/pull/1647/files#diff-c4b9bdd318c6b3aadcb7dbb49cb23d9886d06a08da0d22c1db3a44428db23607R19